shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1190,12 +1185,19 @@ - (void)updateViewportPadding {
 
 #pragma mark - Keyboard events
 
+// using keyboardWillChangeFrame instead of keyboardWill(Show/Hide) because of split/floating
+// keyboard on iPad
 - (void)keyboardWillChangeFrame:(NSNotification*)notification {
+  CGRect screenRect = [[UIScreen mainScreen] bounds];
   NSDictionary* info = [notification userInfo];
+  CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
 
-  // Ignore keyboard notifications related to other apps.
+  // Ignore keyboard notifications related to other apps if not dismissing.
+  bool isDismissing =
+      CGRectEqualToRect(keyboardFrame, CGRectZero) ||
+      keyboardFrame.origin.y >= screenRect.size.height;  // CGRectZero for floating keyboard on iPad
   id isLocal = info[UIKeyboardIsLocalUserInfoKey];
-  if (isLocal && ![isLocal boolValue]) {
+  if (!isDismissing && isLocal && ![isLocal boolValue]) {
```

Comment:
Have you tested If the keyboard is brought be by the other app, does the Flutter app's bottom inset adjust? This logic here seems to prevent Flutter app's bottom inset to be adjusted to the correct value in that case. Which is what I was saying in: https://github.com/flutter/flutter/issues/109845#issuecomment-1222742951

Reply:
As I wrote to @justinmc, if the keyboard is brought by an other app: 

1. the keyboard appear on the other app and the other app is resized
2. (if you tap in the textfield on the flutter app) the textfield in the flutter app becomes the first responder so the flutter app is resized at that moment.

This is the correct behavior since natives apps have that one.